### DISCUSSION TITLE
Calculate acoustic indices for short duration recordings

### DISCUSSION THREAD

**Discussion by p-rocha** [QutEcoacoustics/audio-analysis#547] [Q&A] [answered]
### Task at hand

I'm new to using `AnalysisProgram` and I'm looking to calculate a few acoustic indices from short-duration (10s) wav files. By reading the program's documentation, `audio2csv` seems to be the function I'm looking for. However, this function is meant for long-duration recordings and will attempt to cut my own recordings. 
Forgive my ignorance, but is there any way to skip cutting the recordings into blocks? Even better if I've missed something and there's actually a function that simply calculates the acoustic indices for any given wav file. 

Thanks in advance!

**Comment by atruskie** [comment]
Hi @p-rocha,

The reason we impose a minimum duration is because our noise removal algorithm (for which several off the indices are based on) requires a decently sized sample to operate on.

The indices are also calculated by default for a 60 second resolution... At some point the calculation stops being representative of other minute based indices if the segment is too small and this can lead to subtle but import mistakes in analysis.

Can you explain your use case a little more? Do you really only have 10 second recordings?

**Comment by p-rocha** [comment]
Hi @atruskie, 

Thanks for the swift response. The project I'm working on is meant to compare dB SPL data estimated using various methods (e.g. SPL meter, scikit-maad package) with results from the indices average signal amplitude, signal-to-noise-ratio and background noise, all of which provide data in dB. The restriction of using 10s recordings is because that's the duration of the signal used as control.

This project is primarily exploratory and experimental, but if the results are "accurate", I might model the dB SPL values calculated from longer (30~60s) recordings. In any case, the key issue is being able to calculate acoustic indices straight from short recordings, regardless of their duration, skipping the cutting/subsampling as this would have been "manually" done in previous steps of the analysis. 

ps.: This is part of my PhD project being conducted at QUT under the supervision of Susan Fuller and Paul Roe

**Comment by atruskie** [comment, marked as answer]
As I mentioned already, that would produce indices that are invalid.

One thing you could do is set the IndexCalculationDuration setting to 10 (seconds) and provide the program with minute long segments. You can then extract just the indices result you want for your target 10 seconds.

The reason this works  is we're able to take our noise sample from a wider segment and then calculate indices on only a sub-segment.

**Comment by p-rocha** [comment]
That seems to do the trick! I'll further explore it in the upcoming days.
Thanks for your assistance.

**Reply by towsey** [reply]
Hi @p-rocha,
I am Michael and I worked with Anthony on this code. The main reason for choosing to work with one minute segments was the need for data reduction, i.e. summarising one-minute segments with a few indices potential cuts down the data by a few orders of magnitude.
We chose one-minute because some of the prior publications to our work (eg Sueur and Farina) were calculating indices from one minute segments for the same reason - data reduction.

At some point we decided we wanted to be able to "zoom" in to shorter time intervals. What we discovered that some of the indices work well, notably the traditional signal processing quantities, such as SNR, work well at the shorter time segments but some did not, eg ACI becomes noise at intervals shorter than about 10-15 seconds.

I think Anthony's suggestion is a good one. If you have exactly 10s segments (and it has to be exactly 10 seconds) then you could join six of them together to make a minute and choose 10s as your segment interval. However there is one issue here, to do with the calculation of background noise. We use the modal amplitude to calculate noise which works well over one minute segments. But I think with very short segments you would be better off using the 20th percentile. THis is something you have to experiment with.

I hope this is helpful. But keep in mind you are pushing the software into directions that it was not planned to go!!!

Good luck, Michael

**Reply by p-rocha** [reply]
Hi Michael @towsey,

Thanks for further clarifying the methods and providing further details on the `AnalysisProgram`. 
I did explore @atruskie suggestion, which returned dB values that appear to follow the expected pattern of sound attenuation for the doubling distances. However, the output of the analysis is negative dB values, whereas what I need is absolute dB SPL values comparable with the ones acquired using an SPL meter. 

So I guess you're right and my tentative was indeed pushing the software into directions that it's not meant to go.

In any case, I truly appreciate the support from both of you and I'll keep you posted in case I learn something new. 

Best,
Pedro

**Reply by atruskie** [reply]
As far as I am aware you can't compare decibel values like that - you don't know what gain differences were applied by the sensor hardware before the samples were encoded. 

You'd ideally record a known sound of known intensity at a fixed distance from a microphone and then measure the loudness of that known sound in the recording to calibrate. 

Michael knows more about his than me, but I'll refer you to some slides he recently made on the topic:

- https://openecoacoustics.org/resources/lessons/sound-basics/presentation/#/2
  - the decibel values reported by AP are a ratio of the intensity of the sample
- https://openecoacoustics.org/resources/lessons/sound-basics/presentation/#/3/0/0

**Reply by p-rocha** [reply]
You are absolutely right, acquiring absolute dB SPL from recordings is quite complex and challenging! Thanks for sharing the slides. 
Aside from the method you mentioned (based on a signal of known amplitude), there's an alternative based on mic gain, sensitivity and transducer. I'm currently exploring both alternatives in hopes of estimating dB SPL data from PAM programs. 

Cheers, 

Pedro
